### ISSUE TITLE
Template caching breaks things when NODE_ENV=production

### ISSUE THREAD

**Issue by claviska** [Postleaf/postleaf#39]
Template caching isn't working in production because of a naming conflict with the admin and theme layout file (resolve in https://github.com/Postleaf/empower-theme/commit/04ed6bd163bf83b9272d727bb6ae71715a435e21).

Turns out, there are more issues related to this:

- A race condition exists when using `useThemeViews()` and `useSystemViews()`. A proper renderer needs to be used for admin and theme views to avoid this.
- Adaro doesn't support multiple view folders ([patch submitted](https://github.com/krakenjs/adaro/pull/110)).
- Even with multiple view folders, Dust caching is based on the template's name instead of its location, meaning you can't switch themes without restarting the app. (Dust cache can't be cleared because `dust` is internal to Adaro.)
- Using `chunk.partial()` to render the theme toolbar in `theme_helpers.js` is causing theme views to hang.
- Need to ensure custom error pages still work once a solution is in place.

**Comment by claviska** [comment]
This issue was a 👹 

Due to the way Dust.js caches templates using the template's name instead of its full path, switching themes wasn't possible in production without restarting the app.

One solution I tried was to patch Adaro to support multiple view folders, but again, due to the way Dust.js uses the template name instead of its full path, the problem persisted.

To solve this, I had to create a custom rendering engine for Postleaf that disables both Express caching and Dust.js caching in lieu of its own:

https://github.com/Postleaf/postleaf/blob/master/source/modules/dust_engine.js

The engine could probably be split into its own project at some point. As far as I can tell, it's the only Dust.js engine for Express that supports dynamic views and multiple view folders.

Rolling my own solution wasn't my preferred approach, but after three days of working with Adaro and other existing engines to resolve this, it became necessary. I'm happy to say that template caching now works properly in production. 😎
